scri/bms_transformations.py

```diff
@@ -0,0 +1,829 @@
+        supertranslation[1:4] = sf.vector_as_ell_1_modes(-spacetime_translation[1:4])
+    if "space_translation" in kwargs:
+        s_trans = np.array(kwargs.pop("space_translation"), dtype=float)
+        if s_trans.shape != (3,) or s_trans.dtype != "float":
+            raise TypeError(
+                "\nInput argument `space_translation` should be an array of floats of shape (3,).\n"
+                "Got a {} array of shape {}.".format(s_trans.dtype, s_trans.shape)
+            )
+        spacetime_translation[1:4] = s_trans[:]
+        supertranslation[1:4] = sf.vector_as_ell_1_modes(-spacetime_translation[1:4])
+    if "time_translation" in kwargs:
+        t_trans = kwargs.pop("time_translation")
+        if not isinstance(t_trans, float):
+            raise TypeError("Input argument `time_translation` should be a single float.  " f"Got {t_trans}")
+        spacetime_translation[0] = t_trans
+        supertranslation[0] = sf.constant_as_ell_0_mode(spacetime_translation[0])
+
+    # Decide on the number of points to use in each direction.  A nontrivial supertranslation will
+    # introduce power in higher modes, so for best accuracy, we need to account for that.  But we'll
+    # make it a firm requirement to have enough points to capture the original waveform, at least
+    working_ell_max = 2 * input_ell_max + ell_max_supertranslation
+    if working_ell_max < input_ell_max:
+        raise ValueError(f"working_ell_max={working_ell_max} is too small; it must be at least ell_max={input_ell_max}")
+
+    # Get the rotor for the frame rotation
+    rotation = kwargs.pop("rotation", [1, 0, 0, 0])
+    if type(rotation) is not np.quaternion:
+        rotation = np.quaternion(*np.array(rotation, dtype=float))
+    if rotation.abs() < 3e-16:
+        raise ValueError(f"rotation={rotation} should be a single unit quaternion")
+    rotation = rotation.normalized()
+
+    # Get the boost velocity vector
+    boost = np.array(kwargs.pop("boost", [0.0] * 3), dtype=float)
+    beta = np.linalg.norm(boost)
+    if boost.dtype != float or boost.shape != (3,) or beta >= 1.0:
+        raise ValueError(
+            f"Input boost=`{boost}` should be a 3-vector with " "magnitude strictly less than 1.0"
+        )
+
+    return rotation, boost, supertranslation
+
+def fourvec_to_spin_matrix(fourvec):
+    """Inner product of a four vector and the Pauli matrices,
+    as defined by Eq. (1.2.39) of Spinors and Spacetime Vol. 1
+
+    Parameters
+    ----------
+    fourvec: float array of length 4.
+    """
+    # Maybe better to define Pauli matrices?
+    a = np.cos(fourvec[0] / 2) + 1j * fourvec[3] * np.sin(fourvec[0] / 2)
+    b = (-fourvec[2] + 1j * fourvec[1]) * np.sin(fourvec[0] / 2)
+    c = (+fourvec[2] + 1j * fourvec[1]) * np.sin(fourvec[0] / 2)
+    d = np.cos(fourvec[0] / 2) - 1j * fourvec[3] * np.sin(fourvec[0] / 2)
+
+    return np.array([[a, b], [c, d]])
+
+def Lorentz_to_spin_matrix(Lorentz):
```

Comment:
I thought the pythonic convention was that instances of classes should still start with a lower case letter (though the name of the class itself starts with an upper case letter)